ntoskrnl/ps/i386/psvdm.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Virtual DOS Machine (VDM) support routines
+ * COPYRIGHT:   Copyright Stefan Ginsberg (stefan.ginsberg@reactos.org)
+ *              Copyright 2020 George Bișoc (george.bisoc@reactos.org)
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+ERESOURCE VdmIoListCreationResource;
+
+/* FUNCTIONS *****************************************************************/
+
+/**
+ * @brief
+ * Deletes Virtual DOS Machine (DOS) objects, given an Executive process (EPROCESS).
+ * 
+ * @param[in] Process
+ * A pointer to an Executive process.
+ * 
+ * @return
+ * Nothing.
+ * 
+ */
+VOID
+NTAPI
+PspDeleteVdmObjects(IN PEPROCESS Process)
+{
+    /* If there are no VDM objects, just exit */
+    if (Process->VdmObjects == NULL)
+    {
+        DPRINT1("PspDeleteVdmObjects(): No VDM objects could be found, exiting...\n");
+        return;
+    }
+
+    /* Otherwise return the quota it was taking up and free them */
+    PsReturnProcessNonPagedPoolQuota(Process->VdmObjects, sizeof(Process->VdmObjects));
```

Comment:
The first argument should be the `Process` to charge, not VdmObjects.
For the second argument, `VdmObjects` is a pointer so you'll only get `sizeof(PVOID)` here. You'll need the actual struct size.